bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,913 @@
+            handler.bridgeOffline("No communication with Qbus client");
+        }
+    }
+
+    /**
+     * Cleanup socket when the communication with Qbus Server is closed.
+     *
+     * @throws IOException
+     *
+     */
+    public synchronized void stopCommunication() {
+        listenerStopped = true;
+
+        Socket socket = qSocket;
+
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException ignore) {
+                // ignore IO Error when trying to close the socket if the intention is to close it anyway
+            }
+        }
+
+        qSocket = null;
+
+        CTDConnected = false;
+        logger.debug("Communication stopped from thread {}", Thread.currentThread().getId());
+    }
+
+    /**
+     * Close and restart communication with Qbus Server.
+     *
+     * @throws InterruptedException
+     */
+    public synchronized void restartCommunication() {
+        stopCommunication();
+
+        logger.debug("Qbus: restart communication from thread {}", Thread.currentThread().getId());
+
+        try {
+            startCommunication();
+        } catch (InterruptedException | IOException e) {
+            logger.warn("Error on restaring communication.");
+        }
+    }
+
+    /**
+     * Method to check if communication with Qbus Server is active
+     *
+     * @return True if active
+     */
+    public boolean communicationActive() {
+        return qSocket != null;
+    }
+
+    /**
+     * Method to check if communication with Qbus Client is active
+     *
+     * @return True if active
+     */
+
+    public boolean clientConnected() {
+        return CTDConnected;
+    }
+
+    /**
+     * Runnable that handles incomming communication from Qbus server.
+     * <p>
+     * The thread listens to the TCP socket opened at instantiation of the {@link QbusCommunication} class
+     * and interprets all incomming json messages. It triggers state updates for active channels linked to the
+     * Qbus outputs. It is started after initialization of the communication.
+     *
+     * @return
+     *
+     */
+    private void qEvents() {
+        String qMessage;
+
+        logger.info("Listening for events on thread {}", Thread.currentThread().getId());
+        listenerStopped = false;
+        qEventsRunning = true;
+
+        BufferedReader reader = qIn;
+
+        try {
+            if (reader == null) {
+                throw new IOException("Bufferreader for incomming messages not initialized.");
+            }
+            while (!Thread.currentThread().isInterrupted() & ((qMessage = reader.readLine()) != null)) {
+                if (qMessage != null) {
+                    readMessage(qMessage);
+                }
+            }
+        } catch (IOException e) {
+            if (!listenerStopped) {
+                qEventsRunning = false;
+                logger.error("Qbus: IO error in listener on thread {}", Thread.currentThread().getId());
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging